File: src/Ferrovie/MassaConcreta.java
package Ferrovie;

import java.util.Objects;

public class MassaConcreta extends Massa{
    public MassaConcreta(double m){
        super(m);
    }

    public class MassaFrenata extends Massa{
        double m;
        public MassaFrenata(double m){
            if(this.m<0 || this.m>super.getMassa()){
                throw new IllegalArgumentException("Massa non valida");
            }
            this.m = m;
        }
    }

}

File: Python_Esercizi/Ferrovia/MassaConcreta.py
from Ferrovia import Massa

class MassaConcreta(Massa):
    def __init__(self, m1):
        self._m1 = m1

    def getMassa(self):
        return self._m1

    class MassaFrenata(Massa):
        def __init__(self, m2):
            if(m2 < 0 or m2 > super().getMassa()):
            self._m2 = m2
